### PR TITLE
Exclude files in myfaces-builder-plugin

### PR DESCRIPTION
**Pull request by volosied**
`com.thoughtworks.qdox` doesn't support newer java syntax (i.e `default` on interfaces), and upgrading is out of the question due numerous breaking API changes.

Next best options is to add behavior to exclude processing on specific files. 

Example: 

```
                    <execution>
                        <id>makecomp</id>
                        <configuration>
                            <jsfVersion>20</jsfVersion>
                            <templateComponentName>componentClass20.vm</templateComponentName>
                            <excludes>**/package.html</excludes>
                        </configuration>
                        <goals>
                            <goal>make-components</goal>
                        </goals>
                    </execution>

```
or
```
            <plugin>
                <groupId>org.apache.myfaces.buildtools</groupId>
                <artifactId>myfaces-builder-plugin</artifactId>
                <executions>
                    <execution>
                        <configuration>
                            <excludes>**/src/main/java/jakarta/faces/component/ActionSource.java</excludes>
                        </configuration>
                        <goals>
                            <goal>build-metadata</goal>
                        </goals>
                    </execution>
                    <execution>
                        <id>makecomp</id>
                        <configuration>
                            <jsfVersion>20</jsfVersion>
                            <templateComponentName>componentClass20.vm</templateComponentName>
                            <excludes>**/src/main/java/jakarta/faces/component/ActionSource.java</excludes>
                            <force>true</force>
                        </configuration>
                        <goals>
                            <goal>make-components</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
```

### CODE REVIEW

**Comment by volosied** [comment]
I've been testing MyFaces Core with this snapshot, so we'll need a new release of this plugin. Once that's done, we can make the API changes to pass the sig tests and finally get a final 4.1.0 release out.

File: maven2-plugins/myfaces-builder-plugin/src/main/java/org/apache/myfaces/buildtools/maven2/plugin/builder/BuildMetaDataMojo.java [hunk 1]
```diff
@@ -638,6 +638,7 @@ private Model buildModel(Model model, MavenProject project, ModelParams paramete
         {
             QdoxModelBuilder builder = new QdoxModelBuilder();
             model.setModelId(modelId);
+            System.out.println("VSDEBUG: " + excludes);
```
[the remark this answers is not in the document]
Comment (tandraschko) [reply maven2-plugins/myfaces-builder-plugin/src/main/java/org/apache/myfaces/buildtools/maven2/plugin/builder/BuildMetaDataMojo.java:641]: yeah, please remove

**Comment by tandraschko** [comment]
its ok for me

we should propably remove the builder plugin usage in future from myfaces core completely, and just create the source manually

its stable since years, we dont add, change or remove attributes in general
only once all years

**Comment by volosied** [comment]
Hey Thomas, I agree -- let's tackle it after 4.1

File: maven2-plugins/myfaces-builder-plugin/src/main/java/org/apache/myfaces/buildtools/maven2/plugin/builder/IOUtils.java [hunk 2]
```diff
@@ -440,7 +440,7 @@ private static void getInnerSourceClasses(SourceVisitor visitor,
             }
             else
             {
-                new File((String) i.next());
+                srcDir = new File(dir.toString());
```
Comment (pnicolucci) [inline maven2-plugins/myfaces-builder-plugin/src/main/java/org/apache/myfaces/buildtools/maven2/plugin/builder/IOUtils.java:443]: Thanks for fixing this!

File: maven2-plugins/myfaces-builder-plugin/src/main/java/org/apache/myfaces/buildtools/maven2/plugin/builder/MakeComponentsMojo.java [hunk 3]
```diff
@@ -360,7 +380,10 @@ else if (!outFile.exists())
         for (Iterator i = sourceDirs.iterator(); i.hasNext();)
         {
             String srcDir = (String) i.next();
-            builder.addSourceTree(new File(srcDir));
+            File dir = new File(srcDir);
+            //recusively add files to the builder
```
Comment (pnicolucci) [inline maven2-plugins/myfaces-builder-plugin/src/main/java/org/apache/myfaces/buildtools/maven2/plugin/builder/MakeComponentsMojo.java:384]: Maybe just clean up the formatting of the comment to be consistent but I wouldn't block PR approval on that!

**Review by pnicolucci — APPROVED** [review]
Thanks Vlad!

**Comment by volosied** [comment]
Ran the binary checker to verify nothing else changed (exception ActionSource). It looks good to me. 
image

**Pull request merged**